### PR TITLE
Separate thirdparty dirs

### PR DESCRIPTION
**Pull request by undera**

### CODE REVIEW

File: src/core/org/apache/jmeter/NewDriver.java [hunk 1]
```diff
@@ -85,9 +86,22 @@
...
+        libDirs.add(new File(jmDir + File.separator + "lib")); // $NON-NLS-1$ $NON-NLS-2$
+        libDirs.add(new File(jmDir + File.separator + "lib" + File.separator + "ext")); // $NON-NLS-1$ $NON-NLS-2$
+        libDirs.add(new File(jmDir + File.separator + "lib" + File.separator + "junit")); // $NON-NLS-1$ $NON-NLS-2$
+
+        // add thirdparty extensions to classpath
+        File thirdpartyRoot = new File(jmDir + File.separator + "lib" + File.separator + "3rdparty");
+        File[] thirdpartyDirs = thirdpartyRoot.listFiles(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                return new File(dir, name).isDirectory();
```
Comment (vlsi) [inline src/core/org/apache/jmeter/NewDriver.java]: Please correct to `FileFilter.accept(File pathname)`

**Comment by vlsi** [comment]
@undera , are those merge commits required in this PR?
Can you rebase "Separate thirdparty dirs" commit on top of trunk?

**Comment by undera** [comment]
The git commits won't go into SVN anyway, but I'll try to rebase.

File: src/core/org/apache/jmeter/util/JMeterUtils.java [hunk 2]
```diff
@@ -311,20 +315,47 @@ public void initializeProperties(String file) {
...
+
+        if (thirdpartyDirs != null) {
+            for (File libDir : thirdpartyDirs) {
+                File[] libJars = libDir.listFiles(new FilenameFilter() {
+                    @Override
+                    public boolean accept(File dir, String name) {
+                        return name.endsWith(".jar");// $NON-NLS-1$
+                    }
+                });
+                for (File libJar : libJars) {
```
Comment (vlsi) [inline src/core/org/apache/jmeter/util/JMeterUtils.java]: libJars might throw NPE. Please check for `listFiles != null`

**Pull request closed without merging**